src/os/linux/xdg.rs

```diff
@@ -0,0 +1,412 @@
+                    return Some((parsed, entry_path));
+                }
+            }
+        } else if entry.path().is_dir() {
+            // Recursively keep searching if it is a directory
+            if let Some(result) = find_entry_by_app_name(&entry_path, app_name) {
+                return Some(result);
+            }
+        }
+    }
+    None
+}
+
+fn replace_on_pattern(
+    text: impl AsRef<OsStr>,
+    replace_by: impl AsRef<OsStr>,
+    regex: &Regex,
+) -> OsString {
+    let text = text.as_ref();
+    let replace_by = replace_by.as_ref();
+
+    // Vec<u8> is easier to deal with than OsString, and on unix they're pretty much
+    // the same thing (OsStringExt).
+    let mut result_text = Vec::new();
+    let mut last_index_read = 0;
+
+    for mat in regex.find_iter(text.as_bytes()) {
+        let start = mat.start();
+        let end = mat.end();
+
+        // We put the values from the last index we read, to the start of the matching regex
+        result_text.extend_from_slice(&text.as_bytes()[last_index_read..start]);
+
+        // We put the part we want to replace the match with
+        result_text.extend_from_slice(replace_by.as_bytes());
+
+        // Then we jump the last index to the end of the regex, ignoring the part we matched
+        last_index_read = end;
+    }
+    // At the end of the loop, put the rest of the string
+    result_text.extend_from_slice(&text.as_bytes()[last_index_read..]);
+
+    OsString::from_vec(result_text)
+}
+
+fn parse_quoted_text(
+    text: &OsStr,
+    argument: &OsStr,
+    icon: Option<&OsStr>,
+    desktop_entry_path: Option<&Path>,
+) -> OsString {
+    // We parse the escape character (\) again on the quoted text
+    let mut result = Vec::new();
+    let mut escaping = false;
+    for &c in text.as_bytes() {
+        if escaping {
+            // If escaping, then pass whatever char c is, then stop escaping
+            result.push(c);
+            escaping = false;
+        } else {
+            // If not escaping, check for whether c is escape ('\'), going into escaping
+            // mode if yes (dropping c). Otherwise just pass the c char.
+            if c == b'\\' {
+                escaping = true;
+            } else {
+                result.push(c);
+            }
+        }
+    }
+    let result = OsString::from_vec(result);
+
+    // Now we do the unquoted part
+    parse_unquoted_text(&result, argument, icon, desktop_entry_path)
+}
+
+fn parse_unquoted_text(
+    text: &OsStr,
+    argument: &OsStr,
+    icon: Option<&OsStr>,
+    desktop_entry_path: Option<&Path>,
+) -> OsString {
+    // We parse the arguments
+    // We only have one file path (not an URL). Any instance of these ones
+    // needs to be replaced by the file path in this particular case.
+    let arg_re = byte_regex!(r"%u|%U|%f|%F");
+    let result = replace_on_pattern(text, argument, arg_re);
+
+    // Then the other flags
+    let icon_replace = if let Some(icon_os_str) = icon {
```

Comment:
I believe this expression can be simplified to:
```rust
let icon_replace = icon.unwrap_or_else(|| "".as_ref());
```
(The same goes for `desktop_entry_replace` below)